compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParserErrorHandler.java

```diff
@@ -4706,7 +4707,13 @@ private ParserRuleContext getNextRuleForCloseBrace(int nextLookahead) {
             case SERVICE_DECL:
             case MODULE_CLASS_DEFINITION:
                 endContext();
-                return ParserRuleContext.TOP_LEVEL_NODE;
+                nextToken = this.tokenReader.peek();
```

Comment:
```suggestion
                nextToken = this.tokenReader.peek(nextLookahead);
```

Anyway instead of relying on the next token. Can't we declare an alternative path context say `OPTIONAL_TOP_LEVEL_SEMICOLON`  which has `{ TOP_LEVEL_NODE,  SEMICOLON }`?